include/phasar/PhasarLLVM/DataFlowSolver/Mono/Solver/InterMonoSolver.h

```diff
@@ -344,16 +346,67 @@ template <typename AnalysisDomainTy, unsigned K> class InterMonoSolver {
     }
   }
 
-  mono_container_t
-  summarize(/*Function CalleeTarget, mono_container_t FactAtCall */) {
-    // spawn a new analysis with its own worklist
-  }
+   mono_container_t summarize(f_t CalleeTarget, mono_container_t FactAtCall) {
+    mono_container_t ResultSummary;
+   // auto EntryPoint = ICF->getFunctionName(CalleeTarget); //initialize the entrypoint to the name of CalleeTarget function
+    IntraMonoSolver<AnalysisDomainTy> IMSolver(&IMProblem); //I have problem here creating instance, I tried many ways but recieving error non-matching constructor
+   // providing the reference of IMProblem
+
+    IMSolver.solve(); //calling solve method to analyse the referred problem by IntraMonoSolver
+    
+    auto edges = ICF->getAllControlFlowEdges(CalleeTarget); //ControlflowEdges of CalleeTarget is stored in edges variable
+    for(auto &[Src, Dst]: edges){ //iterate through edges and find the exit statement
+      if(ICF->isExitStmt(Src)){  
+       ResultSummary = IMSolver.getResultsAt(Src);  //get the analysis result of Nth pr return instruction
+      }
+    }
+   return ResultSummary; //return the results
+   }
+
 
-  bool isSensibleToSummarize() {
+
+  bool isSensibleToSummarize(n_t functionNode) {
     // use a heuristic to check whether we should compute a summary
     // make use of the call-graph information
+
+    // first logic to check for leaf functions in a call graph
+     auto isSensible = true;
+    if(ICF->isCallStmt(functionNode)){ // checks if the functionNode is CallStmt
+    auto callsites = ICF->getCallsFromWithin(ICF->getFunctionOf(functionNode)); //get the callsites within the function that is present in the functionNode
+    if(!callsites.empty()){ //if there are callsites, the function is not sensible to summarize and set to false
+      isSensible = false;
+    }
+    }
+    return isSensible; //will consider one among the three logics written here
+     
+    // make use of leafNode function
+  
+    
+  //   auto isSensible = true;
+  //   if(ICF->isCallStmt(functionNode)){ // checks if the functionNode is CallStmt
+  //   for(auto &[Src, Dst] : ICF->getAllControlFlowEdges(ICF->getFunctionOf(functionNode))){ //iterates through the control flow edges of interprocedural function present in functionNode
+  //   if(ICF->isCallStmt(Src)){ //checks for presence of callstmt in the interprocedural function
+  //   isSensible = false; //presence of call statement identifies that the interprocedural function is not a leaf node in call graph and hence not sensible to summarize
+  //   break;
+  //   }
+  //   }
+  //   }
+  //   return isSensible;  
+  // }
   }
 
+  bool isleafFunction(f_t CalleeTarget){
```

Comment:
Does that conceptually differ from your implementation of `isSensibleToSummarize()`?

Reply:
No, the implementation of isleafFunction() and isSensibleToSummarize() is almost the same conceptually but I have used different existing functionality of ICFG to find the leaf/ primitive function.